### PR TITLE
Signup, seeding, doc_version, verification

### PR DESCRIPTION
**Pull request by YuriyTigiev**

### CODE REVIEW

File: controllers/user.controller.ts [hunk 1]
```diff
@@ -27,10 +27,40 @@ class UserController {
       email,
       password,
       role: role || roles[0],
+      isVerified: true,
```
Comment (vicky-gonsalves) [inline controllers/user.controller.ts:30]: I think we need to think a bit here,
When an admin creates the user, isVerified property should be false? and then when user verifies his email we can set it to true? Let me know what you think. 
Also when admin is creating a user, password should be generated automatically with random words and send that password in email to user?

**Comment by vicky-gonsalves** [comment]
Hi Yuriy,
Thanks! Your work is awsome! 
Further we need to add more security to the signup process.
Could you please check this issue: 
#14 
It has complete detail of how the signup process should work.

File: controllers/user.controller.ts (hunk 1 above)
Comment (YuriyTigiev) [reply controllers/user.controller.ts:30]: I thought we should remove the parameter `password` and set `isVerified: false` for this case. Why administrator should set and know a password for a new user? 
What do you think?

Comment (vicky-gonsalves) [reply controllers/user.controller.ts:30]: yes that's exactly what we need to do. So there are 2 user scenarios:
1. When user signs up by himself :  at the begining,  `isVerified` MUST be set to `false` and once he verifies email, `isVerified`  should be set to `true`
2. When Admin adds a user, at the begining,   `isVerified` MUST be set to `false` and password should be generated randomly and save as AES encrypted string in the db. and once User verifies the email, `isVerified`  should be set to `true`

Comment (YuriyTigiev) [reply controllers/user.controller.ts:30]: What about `isDisabled` in both cases?

Comment (vicky-gonsalves) [reply controllers/user.controller.ts:30]: `isDisabled` is a property when we need to disable a user for some time. So it should be set to `false` initially

**Comment by YuriyTigiev** [comment]
> Hi Yuriy,
> Thanks! Your work is awsome!
> Further we need to add more security to the signup process.
> Could you please check this issue:
> #14
> It has complete detail of how the signup process should work.

I will do it. I need time. 

I had done some changes and push them to my repository. When I try to make a new pull request, it shows the previous pull request. Why?

File: helpers/sendmail.helpers.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            from: { email: from_email },
+            content: [
+            { type: "text/html", value: `<h1>Hello ${user}</h1>
+                <br>Please click on following link or Copy/Paste in your web browser to reset your password.
+                <br><a href="http://domain.com/user/reset-password?reset=${token}">http://domain.com/user/reset-password?reset=${token}</a>`
+            },
+            ],
+        };
+        
+        const response = await sendMail(mail, { apiKey: "REDACTED" });
```
Comment (vicky-gonsalves) [inline helpers/sendmail.helpers.ts:33]: Let's keep `apiKey` in the `.env` as it's a config variable.

**Review by vicky-gonsalves — CHANGES_REQUESTED** [review]
Also use `denon fmt` command to format the codebase before the Pull Request.

File: helpers/sendmail.helpers.ts (hunk 2 above)
Comment (YuriyTigiev) [reply helpers/sendmail.helpers.ts:33]: what does mean  apiKey: "REDACTED"
Is not clear.

Comment (vicky-gonsalves) [reply helpers/sendmail.helpers.ts:33]: You should obtain an API key from Sendgrid and paste in there. "REDACTED" is just a random word and not an actual API key.

**Comment by vicky-gonsalves** [comment]
Merged with #19

**Pull request closed without merging**